app/models/user.rb

```diff
@@ -14,11 +14,18 @@ class User < ApplicationRecord
   has_many :user_activities, dependent: :destroy
   has_many :visits, as: :user, dependent: :destroy, inverse_of: :user
   has_one :school_admin, dependent: :restrict_with_error
-  has_many :markdown_attachments, dependent: :restrict_with_error
-  has_many :issued_certificates, dependent: :restrict_with_error
+  has_many :markdown_attachments, dependent: :nullify
+  has_many :issued_certificates, dependent: :nullify
+  has_many :post_likes, dependent: :nullify
+  has_many :text_versions, dependent: :nullify
+  has_many :course_exports, dependent: :nullify
+  has_many :created_posts, class_name: 'Post', foreign_key: 'creator_id', inverse_of: :creator, dependent: :nullify
+  has_many :edited_posts, class_name: 'Post', foreign_key: 'editor_id', inverse_of: :editor, dependent: :nullify
+  has_many :coach_notes, class_name: 'CoachNote', foreign_key: 'author_id', inverse_of: :author, dependent: :nullify
 
   has_secure_token :login_token
   has_secure_token :reset_password_token
+  has_secure_token :delete_account_token
```

Comment:
`has_secure_password` won't actually work, because it makes querying on the table impossible.

```ruby
def delete_account_token
  @delete_account_token_original || super
end

def regenerate_delete_account_token
  @delete_account_token_original = SecureRandom.urlsafe_base64
  update!(delete_account_token: Digest::SHA2.hexdigest(@delete_account_token_original))
end

def find_by_hashed_delete_account_token(delete_account_token)
  find_by(delete_account_token: Digest::SHA2.hexdigest(delete_account_token))
end
```